esphome/components/logger/select/__init__.py

```diff
@@ -21,8 +21,8 @@
 
 
 async def to_code(config):
-    levels = LOG_LEVEL_SEVERITY
-    index = levels.index(CORE.config[CONF_LOGGER][CONF_LEVEL])
+    levels = list(LOG_LEVELS)
+    index = levels.index(CORE.data[CONF_LOGGER][CONF_LEVEL])
     levels = levels[: index + 1]
     var = await select.new_select(config, options=levels)
     await register_parented(var, config[CONF_LOGGER_ID])
```

Comment:
```suggestion
    parent = await cg.get_variable(config[CONF_LOGGER_ID])
    levels = list(LOG_LEVELS)
    index = levels.index(CORE.data[CONF_LOGGER][CONF_LEVEL])
    levels = levels[: index + 1]
    var = await select.new_select(config, options=levels)
    await register_parented(var, parent)
```
Otherwise there is no guarantee that CORE.data[CONF_LOGGER] is set before accessed.

Reply:
Ah yes, good catch